spec/haskell/src/SEL4/Object/TCB/AARCH64.hs

```diff
@@ -51,3 +53,12 @@ getSanitiseRegisterInfo t = do
 
 postModifyRegisters :: PPtr TCB -> PPtr TCB -> UserMonad ()
 postModifyRegisters _ _ = return ()
+
+postSetFlags :: PPtr TCB -> TcbFlags -> Kernel ()
+postSetFlags t flags =
+    when (tcbFlagToWord (ArchFlag FpuDisabled) .&. flags /= 0) (fpuRelease t)
+
+prepareSetDomain :: PPtr TCB -> Domain -> Kernel ()
+prepareSetDomain t newDom = do
```

Comment:
for documentation purposes, might be worth saying why setting domain requires FPU release